nodebuilder/share/config.go

```diff
@@ -1,42 +1,68 @@
 package share
 
 import (
-	disc "github.com/celestiaorg/celestia-node/share/availability/discovery"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/share/availability/discovery"
+	"github.com/celestiaorg/celestia-node/share/availability/light"
 	"github.com/celestiaorg/celestia-node/share/p2p/peers"
 	"github.com/celestiaorg/celestia-node/share/p2p/shrexeds"
 	"github.com/celestiaorg/celestia-node/share/p2p/shrexnd"
 )
 
 type Config struct {
-	// UseShareExchange is a flag toggling the usage of shrex protocols for blocksync.
+	Availability light.Parameters `toml:",omitempty"`
+	Discovery    discovery.Parameters
+
 	UseShareExchange bool
 	// ShrExEDSParams sets shrexeds client and server configuration parameters
 	ShrExEDSParams *shrexeds.Parameters
 	// ShrExNDParams sets shrexnd client and server configuration parameters
 	ShrExNDParams *shrexnd.Parameters
 	// PeerManagerParams sets peer-manager configuration parameters
 	PeerManagerParams peers.Parameters
-	// Discovery sets peer discovery configuration parameters.
-	Discovery disc.Parameters
 }
 
-func DefaultConfig() Config {
-	return Config{
-		UseShareExchange:  true,
+// TODO: Remove share/availability/options.go and reorg configs here
+func DefaultConfig(tp node.Type) Config {
+	cfg := Config{
+		Discovery:         discovery.DefaultParameters(),
 		ShrExEDSParams:    shrexeds.DefaultParameters(),
 		ShrExNDParams:     shrexnd.DefaultParameters(),
+		UseShareExchange:  true,
 		PeerManagerParams: peers.DefaultParameters(),
-		Discovery:         disc.DefaultParameters(),
 	}
+
+	if tp == node.Light {
+		cfg.Availability = light.DefaultParameters()
+	}
```

Comment:
If the value filled only for light node, would it use zero values for all other type of nodes?

Reply:
Yes, and will be omitted in the on-disk config